src/v/utils/unresolved_address.h

```diff
@@ -48,6 +41,8 @@ class unresolved_address {
     uint16_t _port{0};
 };
 
+ss::future<ss::socket_address> resolve_dns(unresolved_address);
```

Comment:
why is this part of unresolved addr. 

also, why is unresolved_addr not part of RPC layer tho.

Reply:
> why is this part of unresolved addr.

I suggested it, for discoverability.
`unresolved_address::resolve()` is essentially the same as `resolve(unresolved_address)`

> also, why is unresolved_addr not part of RPC layer tho.

A fair question

Reply:
it is not used only for RPC.

Reply:
is it a `model::` then ? need more info than `not only used in rpc` tho. what *is* the dependency then.

Reply:
i'd only put the resolving in the RPC layer. I don't think it makes sense for utils to import `rpc` so the `unresolve_address::resolve()` dep cannot work like that. should be  fine  to have `rpc::resolve_dns(addr)`

Reply:
> i'd only put the resolving in the RPC layer. I don't think it makes sense for utils to import `rpc` so the `unresolve_address::resolve()` dep cannot work like that. should be fine to have `rpc::resolve_dns(addr)`

An argument based on dependencies makes sense to me.